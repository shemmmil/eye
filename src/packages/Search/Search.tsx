import React from 'react';
import { Input } from 'antd';

const { Search: InputSearch } = Input;

export const Search = () => {
  return <InputSearch enterButton allowClear />;
};
