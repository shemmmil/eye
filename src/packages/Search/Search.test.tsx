import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Search } from './Search';

type Input = {
  value: string;
};

describe('Search render', () => {
  it('Search work', () => {
    const input = shallow(<Search />);
    expect(toJson(input)).toMatchSnapshot();
  });

  it('Search input correctly render', () => {
    const wrapper = mount(<Search />);

    expect(wrapper.find('.ant-input').length).toBe(1);
  });
  it('Enter symbols correctly work', () => {
    const wrapper = mount(<Search />);
    const input = (wrapper.find('.ant-input') as unknown) as HTMLInputElement;

    input.value = 'Анализ конкурентов';

    expect(input.value).toBe('Анализ конкурентов');
  });
});
