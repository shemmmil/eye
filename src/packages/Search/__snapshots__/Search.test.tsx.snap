// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render Search work 1`] = `
<Search
  allowClear={true}
  enterButton={true}
/>
`;
